packages/config/src/options.js

```diff
@@ -321,16 +321,19 @@ export function getNuxtConfig(_options) {
 
   // include SFCs in node_modules
   options.build.transpile = [].concat(options.build.transpile || [])
-    .map(module => (module instanceof RegExp)
-      ? module
-      : new RegExp(
-        escapeRegExp(
-          path.normalize(
-            module.replace(/\\/g, '/')
-          )
-        )
-      )
-    )
+  const include = []
+  const whitelist = []
+  for (const module of options.build.transpile) {
+    if (module instanceof RegExp) {
+      include.push(module)
+      whitelist.push(module)
+    } else {
+      const posixModule = module.replace(/\\/g, '/')
+      include.push(new RegExp(escapeRegExp(path.normalize(posixModule))))
+      whitelist.push(new RegExp(escapeRegExp(posixModule)))
+    }
+  }
+  Object.assign(options.build.transpile, { include, whitelist })
```

Comment:
Is the type of `build.transpile` an object? Also can't we allow users manually providing their `include`/`whitelists`? I feel changing type of an option is little confusing.

Reply:
include/whitelists should be invisible to users, they only config `transpile`.
We adjust them for users.

Reply:
@clarkdo So if it is an internal resolve mechanism, why don't do it inside `Builder` constructor? Less code execution for `nuxt start` :)

Reply:
It could be, let me think